mlflow/artifacts/__init__.py

```diff
@@ -57,3 +63,114 @@ def download_artifacts(
+            print(file_content)
+
+    .. code-block:: text
+        :caption: Output
+
+        This is a sentence
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_artifact = download_artifacts(artifact_uri, dst_path=tmpdir)
+        with open(local_artifact) as local_artifact_fd:
+            try:
+                return str(local_artifact_fd.read())
+            except Exception as e:
+                _logger.exception(e)
+                raise MlflowException("Unable to form a str object from file content", BAD_REQUEST)
+
+
+def load_dict(artifact_uri: str) -> dict:
+    """
+    Loads content of the artifact file as json/dict object
+    :param artifact_uri: artifact location
+    :return: dict
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+        import json
+        with mlflow.start_run() as run:
+            artifact_uri = run.info.artifact_uri
+            mlflow.log_text(json.dumps({"mlflow-version": "0.28", "n_cores": "10"}), "config.json")
+            config_json = mlflow.artifacts.load_dict(artifact_uri + "/config.json")
+            print(config_json)
+
+    .. code-block:: text
+        :caption: Output
+
+        {'mlflow-version': '0.28', 'n_cores': '10'}
+
+
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_artifact = download_artifacts(artifact_uri, dst_path=tmpdir)
+        with open(local_artifact) as local_artifact_fd:
+            try:
+                return json.load(local_artifact_fd)
+            except json.JSONDecodeError as e:
+                _logger.exception(e)
+                raise MlflowException("Unable to form a JSON object from file content", BAD_REQUEST)
+
+
+def load_image(artifact_uri: str):
+    """
+    Loads content of the artifact file as PIL.Image object
+    :param artifact_uri: artifact location
+    :return: PIL.Image
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+        from PIL import Image
+        with mlflow.start_run() as run:
+            image = Image.new("RGB", (100, 100))
+            artifact_uri = run.info.artifact_uri
+            mlflow.log_image(image, "image.png")
+            image  = mlflow.artifacts.load_image(artifact_uri + "/image.png")
+            print(image)
+
+    .. code-block:: text
+        :caption: Output
+
+        <PIL.PngImagePlugin.PngImageFile image mode=RGB size=100x100 at 0x11D2FA3D0>
+    """
+    try:
+        from PIL import Image
+    except ImportError as exc:
+        raise ImportError(
+            "`load_image` requires Pillow. Please install it via: pip install Pillow"
+        ) from exc
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_artifact = download_artifacts(artifact_uri, dst_path=tmpdir)
+        try:
+            image_obj = Image.open(local_artifact)
+            if is_running_in_ipython_environment():
+                from IPython import display
+
+                display(image_obj)
```

Comment:
Do users **_always_** want to show the loaded image? Maybe not.

Reply:
If this function is called in a loop, an image is displayed in each iteration, which might be annoying.

Reply:
Wait, do we even need to call `IPython.display`?

<img width="433" alt="image" src="https://user-images.githubusercontent.com/17039389/186581005-2460d61b-17f1-48bd-8b49-c1dd4e53101a.png">

Reply:
Test on Jupyter:

<img width="310" alt="image" src="https://user-images.githubusercontent.com/17039389/186583005-ae8cf4c6-c803-4240-8ebb-e159928c554a.png">

Reply:
Test on Databricks

<img width="290" alt="image" src="https://user-images.githubusercontent.com/17039389/186583189-6fdb2a4b-6534-45e6-aedf-0f1b293c034c.png">

Reply:
```suggestion
```

Can we remove this block?

Reply:
@harupy  I tested below code block in notebook. When user calls load_image, image does not get displayed. When `Image.new` is called, yes notebook displays the image but not for `Image.open` which is used in load_image method. according to description of FR, image should be displayed when executed in notebook. I understand your concerns regarding whatif called in a loop, etc. What do you think of changing the function signature to`def load_image(artifact_uri: str, display_image=False)`

```
import mlflow
from PIL import Image
with mlflow.start_run() as run:
    image = Image.new("RGB", (100, 100))
    artifact_uri = run.info.artifact_uri
    mlflow.log_image(image, "image.png")
    image  = mlflow.artifacts.load_image(artifact_uri + "/image.png")
```

Reply:
It's because you assigned the result of `mlflow.artifacts.load_image` to a variable.

<img width="869" alt="image" src="https://user-images.githubusercontent.com/17039389/186816793-67635908-4ddc-4c93-84ab-01d693a8a30f.png">

Reply:
@harupy  Here is the behavior I see, inside a with context manager and one without it. IF users dont call `load_image` within the `with mlflow.start_run()` context, then you are right, they is no need for explicit `IPython.display()`. 

![Screen Shot 2022-08-25 at 10 06 35 PM](https://user-images.githubusercontent.com/1637727/186826364-5bbfa228-4d0d-455e-a3b8-a178193d62b0.png)

Reply:
I think you can:

```pytohn
with mlflow.start_run() as run:
    ...
    image  = mlflow.artifacts.load_image(artifact_uri + "/image.png")

image
```

Reply:
@harupy yes, above suggestion works !! From my side I have pushed PR review changes, please let me know if there is any more details to be taken care of before the merge :)